torchrl/modules/planners/cem.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+
+from torchrl.data.tensordict.tensordict import TensorDictBase
+from torchrl.envs import EnvBase
+from torchrl.modules.planners import MPCPlannerBase
+
+__all__ = ["CEMPlanner"]
+
+
+class CEMPlanner(MPCPlannerBase):
+    """
+    CEMPlanner Module. This class inherits from TensorDictModule.
+
+    Provided a TensorDict, this module will perform a CEM planning step.
+    The CEM planning step is performed by sampling actions from a Gaussian distribution with zero mean and unit variance.
+    The actions are then used to perform a rollout in the environment.
+    The rewards are then used to update the mean and standard deviation of the Gaussian distribution.
+    The mean and standard deviation of the Gaussian distribution are then used to sample actions for the next planning step.
+    The CEM planning step is repeated for a specified number of steps.
+    At the end, we recover the best action which is the one that maximizes the reward given a planning horizon.
+
+    Args:
+        env (Environment): The environment to perform the planning step on (Can be ModelBasedEnv or EnvBase).
+        planning_horizon (int): The number of steps to perform the planning step for.
+        optim_steps (int): The number of steps to perform the MPC planning step for.
+        num_candidates (int): The number of candidates to sample from the Gaussian distribution.
+        num_top_k_candidates (int): The number of top candidates to use to update the mean and standard deviation of the Gaussian distribution.
```

Comment:
don't we want a default value for this and num_candidates (the answer can be no :) )

Reply:
no, unless we want to use PlaNet defaults